src/asynciterable/operators/buffercountwithdebounce.ts

```diff
@@ -0,0 +1,85 @@
+import { OperatorAsyncFunction } from '../../interfaces.js';
+import { AsyncIterableX, concat, of } from '../index.js';
+import { merge } from '../merge.js';
+import { wrapWithAbort } from './withabort.js';
+import { AsyncSink } from '../asyncsink.js';
+import type { bufferCountOrTime } from './buffercountortime.js'; // Used only in jsdoc
+
+const timerEvent = Symbol('BufferCountWithDebounce:TimerEvent');
+const endedEvent = Symbol('BufferCountWithDebounce:EndedEvent');
+
+class BufferCountWithDebounce<TSource> extends AsyncIterableX<TSource[]> {
+  constructor(
+    private readonly _source: AsyncIterable<TSource>,
+    private readonly _bufferSize: number,
+    private readonly _maxWaitTime: number
+  ) {
+    super();
+  }
+
+  async *[Symbol.asyncIterator](signal?: AbortSignal) {
+    const buffer: TSource[] = [];
+
+    const timeoutSink = new AsyncSink<typeof timerEvent>();
+    const merged = merge(concat(this._source, of(endedEvent)), timeoutSink);
+
+    let timeout: NodeJS.Timeout | undefined;
```

Comment:
I don't think we typically use NodeJS types in operators to ensure the types are correct for non-node builds. Could you use the [`sleep`](https://github.com/ReactiveX/IxJS/blob/master/src/asynciterable/_sleep.ts#L3) helper with an abort controller for this instead of `setTimeout()`?

Reply:
You're right, this is nicer. Thanks!